### PR TITLE
state: control dialog visibility with Redux

### PR DESCRIPTION
**Pull request by sjbarag**
Redux allows a more declarative, predictable flow of state throughout an application, and in this case allows dialogs to close themselves. Control the visibility of all dialogs through Redux.

### CODE REVIEW

**Comment by sjbarag** [comment]
Based on #8 , so ignore the first commit here 😃

**Review by otan — APPROVED** [review]

File: src/features/modals/ExportDialog.tsx [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
 interface ExportDialogProps {
-  onHide: () => void;
   show: boolean;
   exportText: string;
   handleSave: () => void;
 }
 
 export const ExportDialog: React.FC<ExportDialogProps> = (props) => {
+  const dispatch = useAppDispatch();
+  const hideModal = useCallback(() => dispatch(modalSlice.actions.hideAll()), []);
```
Comment (nathanstilwell) [inline src/features/modals/ExportDialog.tsx]: Why use `useCallback` here if there are no dependencies?

**Review by nathanstilwell — APPROVED** [review]
One tiny question otherwise, LGTM

File: src/features/modals/ExportDialog.tsx (hunk 1 above)
Comment (sjbarag) [reply src/features/modals/ExportDialog.tsx]: I suppose `dispatch` is a dependency, now that I look at it, eh?  I'll add that to the list.

Basically I didn't want the `hideModal()` callback to be re-generated on every render.

**Pull request merged**